packages/ui/src/helpers/typeGuards.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * Checks if `value` is an Object (excludes arrays and functions)
+ *
+ * @param value
+ * @returns
+ */
+export const isObject = (value: unknown): value is object =>
+  value != null && !Array.isArray(value) && typeof value === 'object';
```

Comment:
Should we align this closer to the [`lodash` implementation](https://github.com/lodash/lodash/blob/master/isObject.js) since that is what is replacing? Same question with `isString` ([`lodash` implementation](https://github.com/lodash/lodash/blob/master/isString.js))

Reply:
plus one, for `isString` especially, because that one will fail if string is initialized through `new String('hi!')`.

This is not at all common practice, but IMO we might as well keep it safe with possible edge cases that lodash has considered.

Reply:
The intent wasn't to replace the lodash function 1:1 but more cover the usecases we have based on current usage. But I hear you..

Reply:
Totally agree with that sentiment for future utils, but have a couple concerns with doing that here:
-  these are replacing `lodash` utils in prod code, if we miss an edge case that `lodash` has accounted for it's our customers that will end up with issues
- future maintainers may not know that these `lodash`-like utils do not cover the same cases as the actual `lodash` versions and may make assumptions based on that

Reply:
Gave it some thought and still feel like these should be tailored to our use cases.
*  The edge cases missing are: `isObject` excludes arrays and functions, `isString` excludes primitive object wrapper which is discouraged to use in the first place. Could you give an example of where these could be an issue in prod code? 
*  Updated jsdoc for each util fn for future maintainers, furthermore unit tests explicitly cover scenarios as fn and array not being objects for the `isObject` util, and string obj wrapper not being a string for purposes of the `isString` util.

Reply:
> The edge cases missing are: isObject excludes arrays and functions, isString excludes primitive object wrapper which is discouraged to use in the first place. Could you give an example of where these could be an issue in prod code?

Since we are a library, we can't assume to know how customers are using our code and whether a subtle change in a utility can cause issues. Additionally just because using `String()` is discouraged, that doesn't mean it won't happen.